internal/pkg/service/common/configmap/bind.go

```diff
@@ -158,40 +157,68 @@ func Bind(cfg BindSpec, targets ...any) error {
 // bind binds flags, ENVs and config files to the configuration structure.
 // Flags have priority over environment variables.
 // The struct is normalized and validated.
-func bind(cfg BindSpec, target ConfigStruct, flags *pflag.FlagSet, flagToFieldFn FlagToFieldFn, configFiles []string) error {
+func bind(cfg BindSpec, target any, flags *pflag.FlagSet, flagToFieldFn FlagToFieldFn, configFiles []string) error {
 	values, err := collectValues(flags, flagToFieldFn, cfg.EnvNaming, cfg.Envs, configFiles)
 	if err != nil {
 		return err
 	}
 
+	// Decode
+	validationErrs := errors.NewMultiError()
 	decoderCfg := &mapstructure.DecoderConfig{
 		TagName:          configKeyTag,
 		ZeroFields:       true,
 		WeaklyTypedInput: true,
 		Result:           target,
 	}
-
 	decoderCfg.DecodeHook = mapstructure.ComposeDecodeHookFunc(
 		unmarshalHook(&decoderCfg.DecodeHook),
 		// additional hooks can be added
 	)
-
 	decoder, err := mapstructure.NewDecoder(decoderCfg)
 	if err != nil {
 		return err
 	}
-
 	if err := decoder.Decode(values); err != nil {
 		return err
 	}
 
-	target.Normalize()
+	// Call Normalize and Validate methods on each value
+	_ = Visit(reflect.ValueOf(target), VisitConfig{
```

Comment:
A tu je hlavna cast implementacie.

Po tom ako sa vsetko namapuje na konfiguracnu strukturu sa vsetky hodnoty do hlbky prejdu:
 - Vola sa `Normalize` a `Validate` ak to daný typ implementuje.